Shard python_functions.cpp

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62188
* #62187
* __->__ #62186
* #62278
* #62185
* #62184

This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

Differential Revision: [D29962046](https://our.internmc.facebook.com/intern/diff/D29962046)

peterbell10 committed:
Shard python_functions.cpp

This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

albanD commented on tools/autograd/templates/python_functions.h:
```diff
@@ -6,6 +6,18 @@
 namespace torch { namespace autograd { namespace generated {
 
-void initialize_autogenerated_functions();
+void initialize_autogenerated_functions_0();
```
Should this be added via a template as well? Otherwise, we should add a comment in the python file where we specify the number of shards explaining that this will need to be updated manually if the number of shards is modified.

peterbell10 replied:
Moved into `gen_autograd_functions.py`, and is now generated by a for-loop.

albanD reviewed (commented):
LGTM
Thanks for working on this!!

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

albanD reviewed (approved):
LGTM

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962046).

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

Differential Revision: [D29962046](https://our.internmc.facebook.com/intern/diff/D29962046)

[ghstack-poisoned]

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

Differential Revision: [D29962046](https://our.internmc.facebook.com/intern/diff/D29962046)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962046).

peterbell10 committed:
Update on "Shard python_functions.cpp"


This file takes 6 minutes on its own to compile and is the limiting factor for
building `libtorch_python` on a 32-core threadripper. This splits the file into
5 shards which take around 50 seconds each to compile.

Differential Revision: [D29962046](https://our.internmc.facebook.com/intern/diff/D29962046)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29962046).
